### ISSUE TITLE
When DatastoreImpl.runQuery()  gets a 503 it does not retry

### ISSUE THREAD

**Issue by simon-temple** [GoogleCloudPlatform/gcloud-java#244]
I'm getting a "Could not reach service" DatastoreException when retrying would run the query successfully.

```
com.google.gcloud.datastore.DatastoreException: Could not reach service
	at com.google.gcloud.datastore.DatastoreException$Code.translate(DatastoreException.java:87)
	at com.google.gcloud.datastore.DatastoreException.translateAndThrow(DatastoreException.java:140)
	at com.google.gcloud.datastore.DatastoreException.translateAndThrow(DatastoreException.java:120)
	at com.google.gcloud.datastore.DatastoreImpl.runQuery(DatastoreImpl.java:113)
	at com.google.gcloud.datastore.QueryResultsImpl.sendRequest(QueryResultsImpl.java:66)
	at com.google.gcloud.datastore.QueryResultsImpl.(QueryResultsImpl.java:56)
	at com.google.gcloud.datastore.DatastoreImpl.run(DatastoreImpl.java:102)
	at com.google.gcloud.datastore.DatastoreImpl.run(DatastoreImpl.java:98)
...
```

**Comment by simon-temple** [comment]
Running 0.0.8-SNAPSHOT

**Comment by mziccard** [comment]
Hi @simon-temple! 
I tried to reproduce the error with a test but it seems to me that the request is being retried.
Could you share with us the code you used to create `DatastoreService` (and its options) and the code that throws the exception? Are you providing some `RetryParams` to `DatastoreOptions`, like:
```java
DatastoreOptions options = DatastoreOptions.builder()
  .retryParams(RetryParams.getDefaultInstance())
  .serviceRpcFactory(rpcFactoryMock)
  .build();
```

**Comment by simon-temple** [comment]
Ah!  Maybe that's because I did not specify RetryParams... if I don't specify them to the options builder do I get NO_RETRIES as default?

```java
PrivateKey pk = PkiUtils.pkcs8PemDecode( pemPrivateKey );
AuthCredentials credentials = AuthCredentials.createFor( clientId, pk );

options = DatastoreOptions.builder()
            .authCredentials( credentials )
            .build();

Datastore datastore = DatastoreFactory.instance().get( options );
```

**Comment by mziccard** [comment]
Yes, if you do not explicitly set `retryParams` the default behaviour is doing no retries. See [here](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-core/src/main/java/com/google/gcloud/ServiceOptions.java#L373).

But we should document this. Please do not close this issue, I'll close it as soon as we add better javadoc.

**Comment by simon-temple** [comment]
Thanks again for your help... I'll start with
```
RetryParams.getDefaultInstance() 
```
and see how we get on.

**Comment by mziccard** [comment]
You're welcome! 
You should not have problems with `getDefaultInstance` but if you need to, you can also build your own retry configuration:
```java
RetryParams.builder()
  .retryMaxAttempts(10)
  .retryMinAttempts(6)
  .maxRetryDelayMillis(30000)
  .totalRetryPeriodMillis(120000)
  .initialRetryDelayMillis(250)
  .build()
```

**Comment by mziccard** [comment]
Closed with #246
